cameraview/src/main/java/com/otaliastudios/cameraview/CameraController.java

```diff
@@ -67,6 +68,7 @@ public void run() {
                 });
             }
         });
+        mFrameManager = new FrameManager(2, this);
```

Comment:
Is 2 frame buffers enough? The CameraSource code uses 4, with the following explanation:

>         // Four frame buffers are needed for working with the camera:
>         //
>         //   one for the frame that is currently being executed upon in doing detection
>         //   one for the next pending frame to process immediately upon completing detection
>         //   two for the frames that the camera uses to populate future preview images

Reply:
I think there's no enough here. This number is related to the max processing time you can afford to spend processing a single frame in order to not skip any future frame.

I can't figure out the exact relation at the moment, but the best number should depend on how quick you are. Without knowing, I would go for 2, because

- 4 buffers would increase this max time but not solve the issue
- If this possible loss is not acceptable you can always use frame = frame.freeze() and spawn another background thread during the process() method
- I don't want to allocate so much memory for people who are not even using processors

Reply:
But in a future PR we could make this configurable, and move the default to 1 or 0 (for who's not using processors). I want people to use the background thread that we provide instead of having to create their own

Reply:
I think adding frame processing as as a boolean flag to `CameraView` would be great. That way people who don't need it, don't suffer any performance hits.

Reply:
It could be automatic, when we have 1 or more processors registered.. But yeah, stuff for another PR